src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = !(this.hidden);
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore
+        && victim.hidden !== true) {
+      victim.health -= 50;
+    }
+
+    if (victim.health <= 0) {
+      victim.health = 0;
+
+      Animal.alive.splice(
+        Animal.alive.findIndex(carcass => carcass === victim), 1);
```

Comment:
if (target.health === 0) {
       Animal.alive = Animal.alive.filter(animal => animal.name !== target.name);
   }

I think you can simplify it!